components/core/Dockerfile

```diff
@@ -1,32 +1,60 @@
-FROM python:3-alpine as builder
+# Container as boiler plate
+FROM python:3.7-slim as base 
 
-WORKDIR /home/Bassa
+# Setting current work directory to /bassa
+WORKDIR /bassa
+
+# Copying API component
 COPY . .
 
-RUN ls && pip install pylint \
-&& pylint -E .
+# Installing dependencies
+RUN apt-get update && \ 
+    apt-get install -y --no-install-recommends \ 
+    python3-pip \ 
+    build-essential \
+    default-libmysqlclient-dev \ 
+    python3-dev \
+    python3-setuptools && \ 
+    rm -rf /var/lib/apt/lists/* && \
+    apt-get clean
+
+# Installing the python modules
+RUN python3 setup.py develop
+
+# -----------------------------
+
+# Container to lint python code
+FROM base as linter 
+
+# Setting current work directory to /Bassa
+WORKDIR /bassa
 
+# Installing pylint and linting the python code
+RUN pip install pylint && pylint -E *.py **/*.py || printf "\n Bassa has got some errors! Continuing the build for now\n\n"
 
-FROM python:3.7
+# -----------------------------
+
+# Container for running the Bassa API server
+FROM base as production
 
 MAINTAINER SCoRe Lab Community <commuity@scorelab.org>
 
 ARG BUILD_DATE
 ARG VCS_REF
 
-WORKDIR /home/Bassa
-EXPOSE 5000
-COPY . .
+# Volume for accessing the downloaded files
+VOLUME [ "/downloads" ]
 
+# Setting current work directory to /Bassa
+WORKDIR /bassa
 
-RUN apt-get update \
-&& ./setup.sh \
-# && python3 setup.py test \ disabing temporary
-&& python3 setup.py develop
-
-CMD [ "./run.sh"]
+# Exposing ports where the server listens 
+EXPOSE 5000
 
+# Start the Bassa API server
+CMD [ "python3", "Main.py" ]
```

Comment:
If we use something like Gunicorn we can have few processes in the container 🤔

Reply:
@JaDogg bro that's a great idea. Shall we have a Gunicorn server for the production Dockerfile? Let's keep this Dockerfile as dev Dockerfile. Let me create a new PR with API integrated with WSGI and nginx. What are your thoughts about it? If yes please approve the PR 😄

Reply:
😂

Reply:
Of course